hack/run-functests.sh

```diff
@@ -1,36 +1,41 @@
 #!/bin/bash
 set -x
-
-# make sure the test runs with specific go vervsion.
-# install go in <ptp-operator-repo>/bin if not already installed
-GO_VERSION=1.20.4
-REPO_BIN_PATH=$(pwd)/bin
-echo "REPO_BIN_PATH is ${REPO_BIN_PATH}"
-export PATH="${REPO_BIN_PATH}/go/bin:$PATH"
-if go version | grep -q "go${GO_VERSION}"; then
-	echo "Go ${GO_VERSION} is already installed."
+# Set go version
+pattern = "/4.1[0-9]/"
+if [[ "$T5CI_VERSION" =~ pattern ]]; then
+  source $HOME/golang-1.20
 else
+  # make sure the test runs with specific go vervsion.
+  # install go in <ptp-operator-repo>/bin if not already installed
+  GO_VERSION=1.20.4
+  REPO_BIN_PATH=$(pwd)/bin
+  echo "REPO_BIN_PATH is ${REPO_BIN_PATH}"
+  export PATH="${REPO_BIN_PATH}/go/bin:$PATH"
+  if go version | grep -q "go${GO_VERSION}"; then
+    echo "Go ${GO_VERSION} is already installed."
+  else
     mkdir -p ${REPO_BIN_PATH}
-	# Check the operating system type
-	if [[ "$(uname)" == "Darwin" ]]; then
-		# macOS
-		GO_BINARY="go${GO_VERSION}.darwin-amd64.tar.gz"
-	elif [[ "$(uname)" == "Linux" ]]; then
-		GO_BINARY="go${GO_VERSION}.linux-amd64.tar.gz"
-	else
-		echo "Unsupported operating system $(uname)."
-		exit 1
-	fi
-	temp_dir=$(mktemp -d)
-	wget https://go.dev/dl/${GO_BINARY} -P "$temp_dir"
-	if uname -a | grep -q "e2e-telco5g"; then
-		# if it is on a Prow job, replace the current go version
-		rm -rf /usr/local/go
-		tar -C /usr/local -xzf "$temp_dir/${GO_BINARY}"
-	else
-		tar -C ${REPO_BIN_PATH} -xzf "$temp_dir/${GO_BINARY}"
-	fi
-	rm -rf "$temp_dir"
+    	# Check the operating system type
+    	if [[ "$(uname)" == "Darwin" ]]; then
+    		# macOS
+    		GO_BINARY="go${GO_VERSION}.darwin-amd64.tar.gz"
+    	elif [[ "$(uname)" == "Linux" ]]; then
+    		GO_BINARY="go${GO_VERSION}.linux-amd64.tar.gz"
+    	else
+    		echo "Unsupported operating system $(uname)."
+    		exit 1
+    	fi
+    	temp_dir=$(mktemp -d)
+    	wget https://go.dev/dl/${GO_BINARY} -P "$temp_dir"
+    	if uname -a | grep -q "e2e-telco5g"; then
```

Comment:
line 30-36 can be replaced by just line 35 since the lines in if statement are meant for running in the runner image which are no longer needed.